server/server.go

```diff
@@ -1077,6 +1078,38 @@ func requestLogger(log *zap.SugaredLogger, config RequestLoggingConfiguration) g
 	}
 }
 
+func extractLoggingMetadata(ctx context.Context) map[string]string {
```

Comment:
it all seems to be ok, but i have a question - since this piece of code is run only once - at the begining of the request, while extract can be called in various contexts, i'm wondering if it would make sense to store already final version of the LoggerFields in the context and in all other places simply rely on that value is already in context? if one needs to add additional fields - it should still be possible with existing api.

Reply:
Yeah, It totally make sense, we can merge requestDetailsKey and loggerContextKey content, and keep both in Context, they will be available to read/write anytime, the only caviat I see is that LoggerFields was originally created to be used in temporal Context, but we can rework on that in a future PR.